src/Marten/Events/Daemon/Coordination/ProjectionCoordinator.cs

```diff
@@ -0,0 +1,252 @@
+        }
+        _runner.SafeDispose();
+
+        foreach (var pair in _daemons)
+        {
+            try
+            {
+                await pair.Value.StopAllAsync().ConfigureAwait(false);
+            }
+            catch (Exception exception)
+            {
+                _logger.LogError(exception, "Error while trying to stop daemon agents in database {Name}", pair.Key.Identifier);
+            }
+        }
+
+        foreach (var daemon in _daemons.Values)
+        {
+            daemon.SafeDispose();
+        }
+
+        try
+        {
+            await Distributor.ReleaseAllLocks().ConfigureAwait(false);
+        }
+        catch (Exception e)
+        {
+            _logger.LogError(e, "Error trying to release subscription agent locks");
+        }
+    }
+
+    private async Task executeAsync(CancellationToken stoppingToken)
+    {
+        await Distributor.RandomWait(stoppingToken).ConfigureAwait(false);
+
+        while (!stoppingToken.IsCancellationRequested)
+        {
+            try
+            {
+                var sets = await Distributor
+                    .BuildDistributionAsync().ConfigureAwait(false);
+
+                foreach (var set in sets)
+                {
+                    // Is it already running here?
+                    if (Distributor.HasLock(set))
+                    {
+                        var daemon = resolveDaemon(set);
+
+                        // check if it's still running
+                        await startAgentsIfNecessaryAsync(set, daemon, stoppingToken).ConfigureAwait(false);
+                    }
+                    else if (await Distributor.TryAttainLockAsync(set, stoppingToken).ConfigureAwait(false))
+                    {
+                        var daemon = resolveDaemon(set);
+
+                        // check if it's still running
+                        await startAgentsIfNecessaryAsync(set, daemon, stoppingToken).ConfigureAwait(false);
+                    }
+                }
+            }
+            catch (Exception e)
+            {
+                if (stoppingToken.IsCancellationRequested) return;
+
+                // Only really expect any errors if there are dynamic tenants in place
+                _logger.LogError(e, "Error trying to resolve projection distributions");
+            }
+
+            if (stoppingToken.IsCancellationRequested) return;
+
+            try
+            {
+                if (_daemons.Values.Any(x => x.HasAnyPaused()))
+                {
+                    await Task.Delay(_options.Projections.AgentPauseTime, stoppingToken).ConfigureAwait(false);
+                }
+                else
+                {
+                    await Task.Delay(_options.Projections.LeadershipPollingTime.Milliseconds(), stoppingToken).ConfigureAwait(false);
+                }
+            }
+            catch (TaskCanceledException)
+            {
+                // just get out of here, this signals a graceful shutdown attempt
```

Comment:
**CONCERN:** Could we add logging here?

Reply:
I honestly don't want to, because this is just indicative of the system being gracefully shutdown. You can get these exceptions even when there's nothing that's actually wrong. I didn't want to log this, because it makes people freak out even though it's really nothing.